apps/app/src/modules/game/dice/dice-zone.tsx

```diff
@@ -54,19 +74,137 @@ export function DiceZone() {
         Launch dice
       </Button>
 
-      <div className={'play'}>
-        <div className={'dices'}>
-          {(currentPlayer?.dices ?? []).map((dice) => (
-            <Dice
-              key={dice.id}
-              dice={dice}
-              selected={lockedDices.includes(dice.id)}
-              onClick={() => onDiceClick(dice)}
-              rotating={!lockedDices.includes(dice.id) && animation}
-            />
-          ))}
+      <DragDropDices
+        setDicesOrder={setDicesOrder}
+        dicesOrder={[...dicesOrder]}
+        setLockedDices={setLockedDices}
+        lockedDices={[...lockedDices]}
+        setIsDragging={setIsDragging}
+      >
+        <div className={cls('h-full', 'flex', 'flex-col', 'justify-around')}>
+          <div className={cls('flex')}>
+            <Droppable
+              droppableId={DiceZoneEnum.playing}
+              direction="horizontal"
+            >
+              {(droppableProvided) => {
+                return (
+                  <div
+                    ref={droppableProvided.innerRef}
+                    className={cls('flex', 'w-full', 'h-[250px]')}
+                    {...droppableProvided.droppableProps}
+                  >
+                    {(
+                      currentPlayer?.dices?.filter(
+                        (dice) => !lockedDices.includes(dice.id)
+                      ) ?? []
+                    )
+                      .sort(
+                        (a, b) =>
+                          dicesOrder.indexOf(a.id) - dicesOrder.indexOf(b.id)
+                      )
+                      .map((dice) => (
```

Comment:
Seeing this, it may make more sens to keep only one of `dicesOrder`/`lockedDices` state and type it as DiceModel[] instead of just the id. This will make this much more simpler to manipulate